torch/_dynamo/variables/constant.py

```diff
@@ -102,7 +102,16 @@ def getitem_const(self, arg: VariableTracker):
 
     @staticmethod
     def is_literal(obj):
-        if type(obj) in (int, float, bool, type(None), str, Ellipsis.__class__):
+        if type(obj) in (
+            int,
+            float,
+            bool,
+            type(None),
+            str,
+            Ellipsis.__class__,
+            torch.dtype,
+            torch.device,
```

Comment:
+ torch.layout

Reply:
don't really want to increase the surface area if not strictly necessary...

Reply:
I mean, the three are the same, there is no reason to not handle them consistently all throughout.

Reply:
yes, but there are plenty of other "literals" (e.g. enums, other torch vars ) that we do not include in this codepath. So I'd prefer to just support the important ones `torch.dtype`, `torch.device`.

See also:
https://github.com/pytorch/pytorch/blob/86196bf116a67701adb0e259cfaa35a843e128ce/torch/_dynamo/guards.py#L386